src/item/enchantment/Enchantment.php

```diff
@@ -34,13 +34,29 @@ class Enchantment{
 	use NotCloneable;
 	use NotSerializable;
 
+	/** @var \Closure(int $level) : int $minCost */
+	private \Closure $minCost;
+
+	/** @var \Closure(int $level, int $minCost) : int $maxCost */
+	private \Closure $maxCost;
+
+	/**
+	 * @phpstan-param null|\Closure(int $level) : int               $minCost
+	 * @phpstan-param null|\Closure(int $level, int $minCost) : int $maxCost
+	 */
 	public function __construct(
 		private Translatable|string $name,
 		private int $rarity,
 		private int $primaryItemFlags,
 		private int $secondaryItemFlags,
-		private int $maxLevel
-	){}
+		private int $maxLevel,
+		?\Closure $minCost = null,
+		?\Closure $maxCost = null,
```

Comment:
The Closures signatures must be checked.

Reply:
What do you mean? Add validation for $minCost and $maxCost passed to the constructor? But why? I don't see similar for other Closure in other classes